Simplenote/SPTextView.swift

```diff
@@ -0,0 +1,130 @@
+import AppKit
+
+// MARK: - SPTextView
+//
+class SPTextView: NSTextView {
+
+    /// Is called when text view resigns being first responder
+    ///
+    var onResignFirstResponder: (() -> Void)?
+
+    /// Highlighted ranges
+    ///
+    var highlightedRanges: [NSRange] = [] {
+        didSet {
+            guard let textStorage = simplenoteStorage else {
+                return
+            }
+
+            textStorage.beginEditing()
+
+            for range in oldValue where range.upperBound <= textStorage.fullRange.upperBound {
+                textStorage.removeAttribute(.backgroundColor, range: range)
+            }
+
+            for range in highlightedRanges {
+                textStorage.addAttribute(.backgroundColor, value: NSColor.simplenoteEditorSearchHighlightColor, range: range)
+            }
+
+            textStorage.endEditingWithoutRestyling()
+        }
+    }
+
+    override var string: String {
+        willSet {
+            // Reset highlights before new string is set
+            highlightedRanges = []
+        }
+    }
+
+    override var typingAttributes: [NSAttributedString.Key : Any] {
+        get {
+            simplenoteStorage?.typingAttributes ?? super.typingAttributes
+        }
+        set { }
```

Comment:
AFAIK this differs from the original behavior. Is this on purpose?

Otherwise maybe:

```
set {
    super.typingAttributes = newValue
}
```

Reply:
You're absolutely right, it is my mistake. Thank you for spotting it!